federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -238,6 +239,17 @@ func GetObjectFromChan(c chan runtime.Object) runtime.Object {
 	}
 }
 
+// GetObjectFromChanExpectNone tries to get an api object from the given channel
+// within a short time window where it's expected that there will not be an event.
+func GetObjectFromChanExpectNone(c chan runtime.Object) runtime.Object {
```

Comment:
+1. This is generally true for all our tests.
It will be great if you can do this. Wont block the PR for it.